### ISSUE TITLE
Unreproducible fsanitize results

### ISSUE THREAD

**Issue by geoffviola** [compiler-explorer/compiler-explorer#1636]
**Describe the bug**
I'm trying out clang's and GCC's sanitizer in compiler explorer to debug undefined behavior in C++ collaboratively. I get different results when I run in the browser versus when I run locally. I'm not sure if there is some state on the server side or if the results are strange because of some internal memory limit.

**To Reproduce**
I have a small program. 
```
#include <array>
#include <cstdint>
#include <cstdlib>
#include <iostream>

int main() {
  std::array<int32_t, 10> arr;
  arr.fill(0);
  std::cout << arr[arr.size()] << std::endl;
  return EXIT_SUCCESS;
}
```
I compile it with GCC or clang with the following options
```
-fsanitize=undefined -fno-sanitize-recover=undefined -O3
```
It shouldn't have any undefined behavior, but the sanitizer is reporting it. When I run locally, I don't see this issue. Both the assembly and the execution windows show that the program exits early and returns 1. When I run the code locally, it prints out normally and returns 0. My local compiler is clang-1100.0.33.8.

**Expected behavior**
Compiler explorer program has the same runtime results as local compiler.

**Link**
Here is the [program](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(j:1,lang:c%2B%2B,source:'%23include+%3Carray%3E%0A%23include+%3Ccstdint%3E%0A%23include+%3Ccstdlib%3E%0A%23include+%3Ciostream%3E%0A%0Aint+main()+%7B%0A++std::array%3Cint32_t,+10%3E+arr%3B%0A++arr.fill(0)%3B%0A++std::cout+%3C%3C+arr%5Barr.size()+-+1%5D+%3C%3C+std::endl%3B%0A++std::cout+%3C%3C+%22hello+world%5Cn%22%3B%0A++return+EXIT_SUCCESS%3B%0A%7D%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:37.857385583913846,l:'4',m:100.00000000000001,n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:clang_trunk,filters:(b:'1',binary:'1',commentOnly:'1',demangle:'1',directives:'1',execute:'1',intel:'1',libraryCode:'1',trim:'1'),lang:c%2B%2B,libs:!(),options:'-fsanitize%3Dundefined+-fno-sanitize-recover%3Dundefined+-O3',source:1),l:'5',n:'0',o:'x86-64+clang+(trunk)+(Editor+%231,+Compiler+%231)+C%2B%2B',t:'0')),header:(),k:34.003639746030395,l:'4',m:100,n:'0',o:'',s:0,t:'0'),(g:!((h:executor,i:(argsPanelShown:'1',compilationPanelShown:'0',compiler:clang_trunk,compilerOutShown:'0',execArgs:'',execStdin:'',lang:c%2B%2B,libs:!(),options:'-fsanitize%3Dundefined+-fno-sanitize-recover%3Dundefined+-O3',source:1,stdinPanelShown:'1'),l:'5',n:'0',o:'x86-64+clang+(trunk)+Executor+(Editor+%231)+C%2B%2B',t:'0')),header:(),k:28.138974670055756,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

**Desktop (please complete the following information):**
 - OS: MacOS 10.15
 - Browser safari
 - Version 13.02

**Comment by partouf** [comment]
Could make different error appear with arr[arr.size()], but noticed the link had -1 like it should have been.
Also tried replacing endl with '\n' because there have been some errors because of that before, but that didn't help in this example. (for example https://godbolt.org/z/KrdlCp produces an error, but replacing std::endl with '\n' fixes that)

Otherwise, this is not reproducible on the local instances with the same Clang and GCC 9.2.0 toolchain, but also other toolchains and clang versions don't work via CE while they do locally.

I suspected firejail, but when I try to reproduce manually on command-line, there's no sanitizer error either. (but maybe I used the wrong command line parameters)

So, I don't know what's going on here.
